### PR TITLE
vmware-guest: Use vmware-vmblock-fuse for drag-and-drop synchronization

### PR DESCRIPTION
**Pull request by kjeremy**
###### Motivation for this change
Fix drag and drop under Gnome on X11.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kjeremy** [comment]
@jshholland Can I get a review please?

**Comment by jshholland** [comment]
In principle LGTM, but I can't do a full review since I'm on holiday all week and so won't be able to test on a VMWare guest until at least Friday night and more realistically the weekend. @joamaki is the maintainer of `open-vm-tools` and so may be a better reviewer this time round.

**Comment by kjeremy** [comment]
@jshholland  @joamaki can I get a review please?

**Review by joamaki — APPROVED** [review]
The change looks reasonable to me, but I can't currently test it right now as I don't have VMware Workstation on my current machine (I'll do a PR to drop me as maintainer for now). @jshholland could you give it a spin if you can? Approving regardless as I don't see an issue with this.

**Comment by Artturin** [comment]
To test this https://nixos.wiki/wiki/Nixpkgs/Reviewing_changes#Modules

**Comment by kjeremy** [comment]
@jshholland any update?

**Comment by kjeremy** [comment]
Can someone help push this through?

**Comment by Artturin** [comment]
Did you test this? if you did then i'll merge it since the changes LGTM

File: nixos/modules/virtualisation/vmware-guest.nix
Comment (Artturin) [inline nixos/modules/virtualisation/vmware-guest.nix:45]: Why umounts.service
thats a target and its added by default to before https://www.freedesktop.org/software/systemd/man/systemd.mount.html#Default%20Dependencies

Comment (kjeremy) [reply nixos/modules/virtualisation/vmware-guest.nix:45]: Good catch. This is based on https://kb.vmware.com/s/article/74671 and I have been running it locally for some time.

**Comment by kjeremy** [comment]
I addressed the comment. I've been running this on my VM since July.

**Comment by dasJ** [comment]
Since #126289 was merged after this PR was opened but before this was merged, it does not evaluate anymore:
```


    error: The option `security.wrappers.vmware-user-suid-wrapper.owner' is used but not defined.

```

**Pull request merged**
